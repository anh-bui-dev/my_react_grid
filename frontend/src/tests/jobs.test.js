import { shallow, mount } from 'enzyme';
import React, { Component } from 'react';
import Jobs from '../components/jobs';
import axios from 'axios';

describe('Jobs', () => {
  it('Should render correctly', () => {
    const wrapper = mount(<Jobs />);
    expect(wrapper).toMatchSnapshot();
  });
    
  it('Call componentDidMount', () => {
    const wrapper = mount(<Jobs />)
    const instance = wrapper.instance();
    jest.spyOn(instance, 'componentDidMount');
    instance.componentDidMount();
    expect(instance.componentDidMount).toHaveBeenCalledTimes(1);
  });

  it('fetches a list', () => {
    const getSpy = jest.spyOn(axios, 'get');
    const wrapper = shallow(<Jobs />);
    setTimeout(() => {
      expect(getSpy).toBeCalled();
    }, 500);
  });
});