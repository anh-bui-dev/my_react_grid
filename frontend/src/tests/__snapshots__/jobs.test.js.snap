// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render correctly 1`] = `
<App>
  <div
    id="container"
  >
    <div
      className="author"
    >
      Jimmy Bui
    </div>
    <Jobs>
      <Loading>
        <div
          className="modal-mask"
        >
          <div
            className="loading d-flex justify-content-center"
          >
            <div
              className="spinner-grow text-secondary"
              role="status"
            >
              <span
                className="sr-only"
              >
                Loading...
              </span>
            </div>
          </div>
        </div>
      </Loading>
    </Jobs>
  </div>
</App>
`;

exports[`Jobs Should render correctly 1`] = `
<Jobs>
  <Loading>
    <div
      className="modal-mask"
    >
      <div
        className="loading d-flex justify-content-center"
      >
        <div
          className="spinner-grow text-secondary"
          role="status"
        >
          <span
            className="sr-only"
          >
            Loading...
          </span>
        </div>
      </div>
    </div>
  </Loading>
</Jobs>
`;
